File: AC/ABC-TDD/144/b.py
n = int(input())
for i in range(1, n+1):
    if i > 9:
        print("No")
        exit()
    if i*i > n:
        print("No")
        exit()
    res = n//i
    if n % i == 0 and 1 <= res and res <= 9:
        print("Yes")
        exit()
print("No")

File: AC/ABC-TDD/144/b.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	for i := 1; i <= n; i++ {
		if i > 9 {
			fmt.Println("No")
			return
		}
		if i*i > n {
			fmt.Println("No")
			return
		}
		var res = n / i
		if n%i == 0 && 1 <= res && res <= 9 {
			fmt.Println("Yes")
			return
		}
	}
	fmt.Println("No")
}
